### DISCUSSION TITLE
How to define the elevation of 2D polylines?

### DISCUSSION THREAD

**Discussion by ilija-lazoroski** [mozman/ezdxf#427] [Q&A] [answered]
I have a shp file with a lot of contours inside and for each of them a fixed elevation, which is looking something like this:

![image](https://user-images.githubusercontent.com/15820737/117813500-7044de00-b263-11eb-8a77-ff993c0ad9b1.png)

Each line inside has elev property.

I want to convert this shape lines into dxf, using Polyline2D with elevation for each contour line.

I have put in an array of tupples [(x, y, z)] and used

`msp.add_polyline2d(contour_plgs, format='v')`

which is giving the lines correctly but I still don't have elevation when viewing the dxf on AutoCAD. I am guessing I need to add an field for each polyline named Elevation, but I don't know how to do that.

Note: I can't use Polyline3D.

Any help will be much appreciated. Sorry If my post is correctly structured.

**Comment by mozman** [comment, marked as answer]
You can define the elevation as vertex, strange - but that is the requirement of the DXF format:

```Python
z = 10  # same z-axis for all vertices
msp.add_polyline2d(contour_plgs, format='v', dxfattibs={
     "elevation": (0, 0, z),
})
```

LWPOLYLINE has a proper elevation attribute:

```Python
z = 10  # same z-axis for all vertices
msp.add_lwpolyline(contour_plgs, format='v', dxfattibs={
     "elevation": z,
})
```

Other solution: translate entity in z-direction:
```Python
z = 10  # same z-axis for all vertices
entity = msp.add_polyline2d(contour_plgs, format='v')
entity.translate(0, 0, z)
```

**Reply by ilija-lazoroski** [reply]
Thank you a lot. This is want I needed.
